### PR TITLE
Add documentation to TASTy reflect

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/reflect/Core.scala [hunk 1]
```diff
@@ -126,17 +126,40 @@ trait Core {
 
   /** Tree representing executable code written in the source */
   type Tree
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/Core.scala]: I'd remove ~executable~ from the doc above.

File: library/src/scala/tasty/reflect/FlagSet.scala [hunk 2]
```diff
@@ -1,31 +1,85 @@
...
+  /** Is this symbol `implicit`. */
   def isImplicit: Boolean
+
+  /** Is this symbol `erased`. */
   def isErased: Boolean
+
+  /** Is this symbol `lazy`. */
   def isLazy: Boolean
+
+  /** Is this symbol `override`. */
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/FlagSet.scala]: Maybe be consistent here to remove the period?

File: library/src/scala/tasty/reflect/FlagSet.scala [hunk 3]
```diff
@@ -1,31 +1,85 @@
...
+  /** Is this symbol marked as static. Mapped to static Java member. */
+  def isStatic: Boolean
+
+  /** Is this symbol an object or its class (used for a ValDef or a ClassDef extends Modifier respectively) */
+  def isObject: Boolean
+
+  /** Is this symbol a trait. */
+  def isTrait: Boolean
+
+  /** Is this symbol local. Used in conjunction with Private/private[Type] to mean private[this] extends Modifier proctected[this]. */
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/FlagSet.scala]: Change the period to `?`

File: library/src/scala/tasty/reflect/Core.scala [hunk 4]
```diff
@@ -124,19 +124,42 @@ trait Core {
   /** Workaround missing `|` types in Scala 2 to represent `Term | TypeTree` */
   type TermOrTypeTree /* Term | TypeTree */
 
-  /** Tree representing executable code written in the source */
+  /** Tree representing code written in the source */
   type Tree
+
+    /** Tree representing a pacakage clause in the source code */
```
Comment (ShaneDelmore) [inline library/src/scala/tasty/reflect/Core.scala]: /s/pacakage/package

**Pull request merged**